src/mqt/benchviewer/backend.py

```diff
@@ -23,256 +23,548 @@
+                & (self.database["num_qubits"] <= max_qubits)
+                & (self.database["benchmark"].isin(selected_scalable_benchmarks))
+            )
+            | (self.database["benchmark"].isin(selected_nonscalable_benchmarks))
+        ]
+
+        if indep_qiskit_compiler:
+            db_tmp1 = db_tmp.loc[(db_tmp["indep_flag"]) & (db_tmp["compiler"] == "qiskit")]
+            db_filtered = pd.concat([db_filtered, db_tmp1])
+
+        if indep_tket_compiler:
+            db_tmp2 = db_tmp.loc[(db_tmp["indep_flag"]) & (db_tmp["compiler"] == "tket")]
+            db_filtered = pd.concat([db_filtered, db_tmp2])
+
+        if nativegates_qiskit_compiler:
+            for gate_set in native_gatesets:
+                for opt_lvl in native_qiskit_opt_lvls:
+                    db_tmp3 = db_tmp.loc[
+                        (db_tmp["nativegates_flag"])
+                        & (db_tmp["gate_set"] == gate_set)
+                        & (db_tmp["compiler"] == "qiskit")
+                        & (db_tmp["compiler_settings"] == opt_lvl)
+                    ]
+                    db_filtered = pd.concat([db_filtered, db_tmp3])
+
+        if nativegates_tket_compiler:
+            for gate_set in native_gatesets:
+                db_tmp4 = db_tmp.loc[
+                    (db_tmp["nativegates_flag"]) & (db_tmp["gate_set"] == gate_set) & (db_tmp["compiler"] == "tket")
+                ]
+                db_filtered = pd.concat([db_filtered, db_tmp4])
+
+        if mapped_qiskit_compiler:
+            for opt_lvl in mapped_qiskit_opt_lvls:
+                for device in mapped_devices:
+                    db_tmp5 = db_tmp.loc[
+                        (db_tmp["mapped_flag"])
+                        & (db_tmp["target_device"] == device)
+                        & (db_tmp["compiler"] == "qiskit")
+                        & (db_tmp["compiler_settings"] == opt_lvl)
+                    ]
+                    db_filtered = pd.concat([db_filtered, db_tmp5])
+
+        if mapped_tket_compiler:
+            for placement in mapped_tket_placements:
+                for device in mapped_devices:
+                    db_tmp6 = db_tmp.loc[
+                        (db_tmp["mapped_flag"])
+                        & (db_tmp["target_device"] == device)
+                        & (db_tmp["compiler"] == "tket")
+                        & (db_tmp["compiler_settings"] == placement)
+                    ]
+                    db_filtered = pd.concat([db_filtered, db_tmp6])
+
+        return db_filtered["path"].to_list()
+
+    class NoSeekBytesIO:
+        def __init__(self, fp: io.BytesIO):
+            self.fp = fp
+            self.deleted_offset = 0
+
+        def write(self, b):
+            return self.fp.write(b)
+
+        def tell(self):
+            return self.deleted_offset + self.fp.tell()
+
+        def hidden_tell(self):
+            return self.fp.tell()
+
+        def seekable(self):
+            return False
```

Comment:
Could we test these lines as well to make sure that ZIP creation works properly?

Reply:
I added a test which streams files from the zip for both a valid and an invalid case.